src/color.cc

```diff
@@ -213,6 +213,47 @@ ColorModule::~ColorModule()
 {
 }
 
+// Parses hex colors according to: https://drafts.csswg.org/css-color/#typedef-hex-color.
+// If the input is invalid, returns boost::none.
+// Supports the following formats:
+// * "#rrggbb"
+// * "#rrggbbaa"
+// * "#rgb"
+// * "#rgba"
+boost::optional<Color4f> parse_hex_color(const std::string& hex) {
+	// validate size. short syntax uses one hex digit per color channel instead of 2.
+	const bool short_syntax = hex.size() == 4 || hex.size() == 5;
+	const bool long_syntax = hex.size() == 7 || hex.size() == 9;
+	if (!short_syntax && !long_syntax) return boost::none;
+
+	if (hex[0] != '#') return boost::none;
+
+	// number of characters per color channel
+	const int stride = short_syntax ? 1 : 2;
+	const float channel_max = short_syntax ? 15 : 255;
+
+	std::array<int,4> rgba;
```

Comment:
For a small improvement in readability, what if you made this a `Color4f`, and in the end just returned `rgba / channel_max`?

Reply:
Cool, I didn't see that `Color4f` was just an `Eigen::Vector`.